File: ras_temp.js
const fetch = require('node-fetch'); // You may need to install the 'node-fetch' package

function convertKelvinToCelsius(temp) {
    return temp - 273.15;
}

async function getWeatherData(lat, lon, key) {
    const apilinkCurrent = `https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${key}`;
    const apilinkForecast = `https://api.openweathermap.org/data/2.5/forecast?lat=${lat}&lon=${lon}&appid=${key}`;

    const responseForecast = await fetch(apilinkForecast);
    const responseCurrent = await fetch(apilinkCurrent);

    const jsonFor = await responseForecast.json();
    const jsonCur = await responseCurrent.json();

    const globalWeatherCur = jsonCur.weather[0].main;
    const tempCur = convertKelvinToCelsius(jsonCur.main.temp_max);

    console.log("Le minimum du jour : " + getMinDay(jsonFor) + "*C");
    console.log("Le maximum du jour : " + getMaxDay(jsonFor) + "*C");
    console.log("Le temps actuel : " + globalWeatherCur);
    console.log("La temperature actuelle : " + tempCur + "*C");
    
    return {
        "min": getMinDay(jsonFor),
        "max": getMaxDay(jsonFor),
        "current": tempCur,
        "weather": globalWeatherCur
    };
}

function getMinDay(jsonFor) {
    let minDay = -100000;
    for (let i = 0; i < 12; i++) {
        if (jsonFor.list[i].main.temp_min > minDay) {
            minDay = jsonFor.list[i].main.temp_min;
        }
    }
    return parseInt(convertKelvinToCelsius(minDay));
}

function getMaxDay(jsonFor) {
    let maxDay = 100000;
    for (let i = 0; i < 12; i++) {
        if (jsonFor.list[i].main.temp_max < maxDay) {
            maxDay = jsonFor.list[i].main.temp_max;
        }
    }
    return parseInt(convertKelvinToCelsius(maxDay));
}

const lat = 45.764671;
const lon = 4.8804;
const language = "en";
const key = "bdeb8474ac275502254ef9113a922598";

getWeatherData(lat, lon, key);

//Make the api call every 10 minutes
setInterval(function () {
    getWeatherData(lat, lon, key);
}, 600000);

File: ras_temp.py
import requests


def convert_kelvin_to_celsius(temp):
    return temp - 273.15

def get_max_day(jsonfor):
    min_day= -100000
    for i in range(0, 12):
        if jsonfor["list"][i]["main"]["temp_min"] > min_day:
            min_day = jsonfor["list"][i]["main"]["temp_min"]
    return int(convert_kelvin_to_celsius(min_day))

def get_min_day(jsonfor):
    max_day= 100000
    for i in range(0, 12):
        if jsonfor["list"][i]["main"]["temp_max"] < max_day:
            max_day = jsonfor["list"][i]["main"]["temp_max"]
    return int(convert_kelvin_to_celsius(max_day))

lat = 45.764671
lon = 4.880400
language = "en"
key = "bdeb8474ac275502254ef9113a922598"
apilinkcurrent = f"https://api.openweathermap.org/data/2.5/weather?lat={lat}&lon={lon}&appid={key}"
apilinkforecast = f"https://api.openweathermap.org/data/2.5/forecast?lat={lat}&lon={lon}&appid={key}"

jsonfor = requests.get(apilinkforecast).json()
jsoncur = requests.get(apilinkcurrent).json()


global_weather_cur = jsoncur["weather"][0]["main"]
temp_cur = int(convert_kelvin_to_celsius(jsoncur["main"]["temp_max"]))

print("Le minimum du jour : " + str(get_min_day(jsonfor)) + "*C")
print("Le maximum du jour : " + str(get_max_day(jsonfor)) + "*C")
print("Le temps actuel : " + str(global_weather_cur))
print("La temperature actuelle : " + str(temp_cur) + "*C")

